repostats/repostats.py

```diff
@@ -1082,10 +1090,50 @@ def get_astroconda_membership(name="", data=""):
     return False
 
 
+def scrap_rtd_badge(repoorg, reponame):
+    import base64
+    import re
+
+    content = None
+    badge = None
+
+    # TODO: Add more possibilities as needed.
+    readme_files = ('README', 'README.md', 'README.rst', 'README.txt')
+
+    for filename in readme_files:
+        url = (_repo_base.format(repoorg, reponame) +
+               '/contents/{}'.format(filename))
+        try:
+            json = get_api_data(url)
+            if json is not None:
+                content = base64.b64decode(json['content']).decode('utf-8')
+        except Exception:
+            pass
+        else:
+            if content is not None:
+                break
+
+    if content is None:
+        return badge
+
+    # Regex magic by Craig Jones.
+    m = re.search('(http[s]:\/\/readthedocs.*version=[\w]+)', content)
+    if m is None:
+        return badge
+    badge = m.group(0)
+
+    return badge
```

Comment:
Can we restructure this for loop to be a little more clear? Perhaps something like this:
```
for filename in readme_files:
     url = (_repo_base.format(org, name) +
              '/contents/{}'.format(filename))
          
     json = get_api_data(url)
     if json is not None:  # we found the right readme name
         content = base64.b64decode(json['content']).decode('utf-8')   # decipher
         if content is not None:  # readme is populated
             m = re.search('(http[s]:\/\/readthedocs.*version=[\w]+)', content) # Regex magic by Craig Jones.
             if m is not None:  #theres an RTD listing
                 badge = m.group(0)
             return badge  #so sad
return badge  #should be None
```
I  think this covers all the cases..?

Reply:
Sure. I didn't spend too much time cleaning the logic flow.

Reply:
cool. Just run a quick check to make sure it still works as you expect?

Reply:
Yes, I did. At least for that 4 repos I tested.